wasi_snapshot_preview1.random_get nil pointer dereference (Go 1.21rc2 built application panics)

**Describe the bug**

A basic application build with Go 1.21rc2 when loaded into the wazero runtime panics when the `random_get` wasi host function is called.

**To Reproduce**

Build a wasm that doesn't do anything:

```go
package main
func main() { }
```

```
GOOS=wasip1 GOARCH=wasm go build -o main.wasm main.go
```

Build a host using wazero v1.2.1 a host that supports wasip1:

```go
package main

import (
	"context"
	_ "embed"
	"fmt"

	"github.com/tetratelabs/wazero"
	"github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1"
)

//go:embed main.wasm
var wasm []byte

func main() {
	ctx := context.Background()

	r := wazero.NewRuntime(ctx)
	defer r.Close(ctx)

	_, err := wasi_snapshot_preview1.Instantiate(ctx, r)
	if err != nil {
		panic(err)
	}

	mod, err := r.Instantiate(ctx, wasm)
	if err != nil {
		panic(err)
	}

	f := mod.ExportedFunction("_start")
	if f != nil {
		res, err := f.Call(ctx)
		if err != nil {
			panic(err)
		}
		fmt.Println("start =", res)
	}
}
```

Run the host Go application.

**Expected behavior**

The application to execute.

**Actual behavior**
```
go run ./host
panic: runtime error: invalid memory address or nil pointer dereference (recovered by wazero)
wasm stack trace:
        wasi_snapshot_preview1.random_get(i32,i32) i32
        .runtime.random_get(i32) i32
        .runtime.getRandomData(i32) i32
        .runtime.alginit(i32) i32
        .runtime.schedinit(i32) i32
        .runtime.rt0_go(i32) i32
        ._rt0_wasm_wasip1()

Go runtime stack trace:
goroutine 1 [running]:
runtime/debug.Stack()
        /Users/leighmcculloch/.local/bin/go/latest/src/runtime/debug/stack.go:24 +0x64
github.com/tetratelabs/wazero/internal/wasmdebug.(*stackTrace).FromRecovered(0x140004277e0?, {0x100a6dcc0?, 0x100d1d8a0?})
        /Users/leighmcculloch/.local/gopath/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/internal/wasmdebug/debug.go:139 +0xc4
github.com/tetratelabs/wazero/internal/engine/compiler.(*callEngine).deferredOnCall(0x14000000360, {0x100aa3740, 0x100d5cbe0}, 0x100aa5358?, {0x100a6dcc0, 0x100d1d8a0})
        /Users/leighmcculloch/.local/gopath/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/internal/engine/compiler/engine.go:855 +0x37c
github.com/tetratelabs/wazero/internal/engine/compiler.(*callEngine).call.func1()
        /Users/leighmcculloch/.local/gopath/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/internal/engine/compiler/engine.go:732 +0x68
panic({0x100a6dcc0?, 0x100d1d8a0?})
        /Users/leighmcculloch/.local/bin/go/latest/src/runtime/panic.go:914 +0x218
github.com/tetratelabs/wazero/internal/sys.(*Context).RandSource(...)
        /Users/leighmcculloch/.local/gopath/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/internal/sys/sys.go:109
github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1.randomGetFn({0x0?, 0x100d567b0?}, {0x100aa5358?, 0x14000163a40?}, {0x14000134110?, 0x100a72c40?, 0x14000427c01?})
        /Users/leighmcculloch/.local/gopath/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/imports/wasi_snapshot_preview1/random.go:41 +0x3c
github.com/tetratelabs/wazero/imports/wasi_snapshot_preview1.wasiFunc.Call(0x1289dc3e8?, {0x100aa3740?, 0x100d5cbe0?}, {0x100aa5358?, 0x14000163a40?}, {0x14000134110, 0x2, 0x0?})
        /Users/leighmcculloch/.local/gopath/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/imports/wasi_snapshot_preview1/wasi.go:293 +0x58
github.com/tetratelabs/wazero/internal/engine/compiler.(*callEngine).execWasmFunction(0x14000000360, {0x100aa3740, 0x100d5cbe0}, 0x100e5c108?)
        /Users/leighmcculloch/.local/gopath/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/internal/engine/compiler/engine.go:1024 +0x118
github.com/tetratelabs/wazero/internal/engine/compiler.(*callEngine).call(0x14000000360, {0x100aa3740, 0x100d5cbe0}, {0x0?, 0x14000427e48?, 0x10099f84c?}, {0x0, 0x0, 0x0})
        /Users/leighmcculloch/.local/gopath/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/internal/engine/compiler/engine.go:747 +0x224
github.com/tetratelabs/wazero/internal/engine/compiler.(*callEngine).Call(0x14000163a40?, {0x100aa3740?, 0x100d5cbe0?}, {0x0?, 0x153849?, 0x153849?})
        /Users/leighmcculloch/.local/gopath/pkg/mod/github.com/tetratelabs/wazero@v1.2.1/internal/engine/compiler/engine.go:701 +0xc0
main.main()
        /Users/leighmcculloch/Code/wazero-play/host/main.go:33 +0x13c


goroutine 1 [running]:
main.main()
        /Users/leighmcculloch/Code/wazero-play/host/main.go:35 +0x1c4
```

**Screenshots**
N/A

**Environment (please complete the relevant information):**
- Go version: 1.21rc2
- wazero Version: v1.2.1
- Host architecture: macOS arm64
- Runtime mode: interpreter

**Additional context**
Add any other context about the problem here.

--- a/config.go
+++ b/config.go
@@ -493,7 +493,7 @@ type ModuleConfig interface {
 	// # Notes
 	//
 	//   - If a start function doesn't exist, it is skipped. However, any that
-	//	   do exist are called in order.
+	//     do exist are called in order.
 	//   - Start functions are not intended to be called multiple times.
 	//     Functions that should be called multiple times should be invoked
 	//     manually via api.Module's `ExportedFunction` method.

--- a/config.go
+++ b/config.go
@@ -492,10 +492,14 @@ type ModuleConfig interface {
 	//
 	// # Notes
 	//
-	//   - If any function doesn't exist, it is skipped. However, all functions
-	//	  that do exist are called in order.
-	//   - Some start functions may exit the module during instantiate with a
-	//	  sys.ExitError (e.g. emscripten), preventing use of exported functions.
+	//   - If a start function doesn't exist, it is skipped. However, any that
+	//	   do exist are called in order.
+	//   - Start functions are not intended to be called multiple times.
+	//     Functions that should be called multiple times should be invoked
+	//     manually via api.Module's `ExportedFunction` method.
+	//   - Start functions commonly exit the module during instantiation,
+	//     preventing use of any functions later. This is the case in "wasip1",
+	//     which defines the default value "_start".
 	WithStartFunctions(...string) ModuleConfig
 
 	// WithStderr configures where standard error (file descriptor 2) is written. Defaults to io.Discard.

--- a/runtime.go
+++ b/runtime.go
@@ -31,7 +31,8 @@ import (
 //   - Closing this closes any CompiledModule or Module it instantiated.
 type Runtime interface {
 	// Instantiate instantiates a module from the WebAssembly binary (%.wasm)
-	// with default configuration.
+	// with default configuration, which notably calls the "_start" function,
+	// if it exists.
 	//
 	// Here's an example:
 	//	ctx := context.Background()

